### PR TITLE
Optional goody: Convert class components to functional components. 

### PR DESCRIPTION
**Pull request by zhuyuanmao**
Converted all class components to functional components #4, and replaced all `connect` API to Hooks, such as `useSelector` and `useDispatch`. So the front-end codebase will be cleaner and easy to read than the one with `connect` API.  
* Updated material-ui components from deprecated ones to the newest ones.  
* Fixed bolding preview text bug that the sender was able to see a bolding preview text he sent.  
* Removed unused dependencies.

### CODE REVIEW

File: client/src/Login.js [hunk 1]
```diff
@@ -13,22 +13,23 @@ import { login } from "./store/utils/thunkCreators";
 
 const Login = (props) => {
   const history = useHistory();
-  const { user, login } = props;
+  const user = useSelector(state => state.user);
+  const dispatch = useDispatch();
```
Comment (JeffrinManovaPrabahar) [inline client/src/Login.js:17]: Nice use of useSelector & useDispatch here 👍

File: client/src/components/ActiveChat/ActiveChat.js [hunk 2]
```diff
@@ -53,15 +53,10 @@ const ActiveChat = (props) => {
...
-    user: state.user,
-    conversation:
-      state.conversations &&
-      state.conversations.find(
-        (conversation) => conversation.otherUser.username === state.activeConversation
-      )
-  };
-};
+const getConversation = (state) => {
+  return state.conversations && state.conversations.find(
```
Comment (JeffrinManovaPrabahar) [inline client/src/components/ActiveChat/ActiveChat.js:57]: Nice converstion.

File: client/src/components/ActiveChat/Input.js [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
-import { withStyles } from "@material-ui/core/styles";
-import { connect } from "react-redux";
-import { postMessage } from "../../store/utils/thunkCreators";
-import { readMessages } from "../../store/utils/thunkCreators";
+import {  useDispatch, useSelector } from "react-redux";
+import { postMessage,readMessages } from "../../store/utils/thunkCreators";
+import { makeStyles } from "@material-ui/core/styles";
 
-const styles = {
+const useStyles = makeStyles((theme) => ({
```
Comment (JeffrinManovaPrabahar) [inline client/src/components/ActiveChat/Input.js:7]: Good job in converting this to useStyles.

File: client/src/components/ActiveChat/Input.js [hunk 4]
```diff
@@ -1,11 +1,10 @@
...
   root: {
     justifySelf: "flex-end",
     marginTop: 15,
```
Comment (JeffrinManovaPrabahar) [inline client/src/components/ActiveChat/Input.js]: `theme` gets passed as an argument to `makeStyles` and can be used for styles like font, margin, padding, colours (e.g. theme.spacing for padding/margins, theme.palette for colours) to help keep styling consistent across the app.

File: client/src/components/Home.js [hunk 5]
```diff
@@ -1,86 +1,54 @@
...
+  const classes = useStyles();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const user = useSelector(state => state.user);
+  const dispatch = useDispatch();
+  const userId = useRef(user.id);
+
+  useEffect(() => {
+    if (userId.current !== userId) {
+      setIsLoggedIn(true);
     }
```
Comment (JeffrinManovaPrabahar) [inline client/src/components/Home.js]: Here you have compared the ref value with the ref itself, which doesn't make sense. Can you check this again?

**Review by JeffrinManovaPrabahar — COMMENTED** [review]
Good job overall in converting class components to functional components.

I have mentioned few issues in below comments, please take a look at them.

Please let us know on Slack when you have addressed the issues and this is ready to be reviewed again.

File: client/src/components/ActiveChat/Input.js (hunk 4 above)
Comment (FilipeGorodscy) [reply client/src/components/ActiveChat/Input.js]: Looks like this comment wasn't addressed :)

File: client/src/components/Sidebar/ChatContent.js [hunk 6]
```diff
@@ -28,24 +29,29 @@ const useStyles = makeStyles((theme) => ({
...
-};
-
+}
+const unreadByCurrentUser = (user, conversation) => {
+  const lastMsg = conversation.messages[conversation.messages.length-1];
+  if (lastMsg.senderId === user.id || lastMsg?.readStatus){
+    return false;
+  }
+  return true;
+}
```
Comment (FilipeGorodscy) [inline client/src/components/Sidebar/ChatContent.js]:
```suggestion
  return lastMsg.senderId !== user.id && !lastMsg?.readStatus
```

**Review by FilipeGorodscy — APPROVED** [review]
Looks like some of the previous comments weren't addressed, please double check :) Feel free to merge after the changes.

**Pull request merged**